core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/AbstractAws4Signer.java

```diff
@@ -426,7 +379,7 @@ private long getSignatureDurationInSeconds(Aws4SignerRequestParams requestParams
     private byte[] newSigningKey(AwsCredentials credentials,
                                  String dateStamp, String regionName, String serviceName) {
         byte[] kSecret = ("AWS4" + credentials.secretAccessKey())
-            .getBytes(Charset.forName("UTF-8"));
+            .getBytes(StandardCharsets.UTF_8);
```

Comment:
Does this needs to be recomputed all of the time? Can't the credentials holds kSecret?

Reply:
That's a cool idea. We'd probably want to use a weak hash map from credential to cached values for that credential so that we don't leak that information out of the signer, but that could definitely save some computation for reused credentials.

Reply:
I thought `credentials` was immutable when commenting. If mutability is in the game it would ruin potentially ruin gains...